src/api/app/models/workflow/step.rb

```diff
@@ -37,16 +37,26 @@ def create_or_update_subscriptions(package, workflow_filters)
 
   protected
 
-  def validate_step_instructions
+  def validate_step_keys
     self.class::REQUIRED_KEYS.each do |required_key|
       errors.add(:base, "The '#{required_key}' key is missing") unless step_instructions.key?(required_key)
     end
   end
 
+  def validate_step_values
+    self.class::REQUIRED_KEYS.each do |required_key|
+      errors.add(:base, "The '#{required_key}' key must provide a value") if step_instructions[required_key].blank?
```

Comment:
If `required_key` isn't present in `step_instructions`, `step_instructions[required_key].blank?` will cause the following exception:

```
NoMethodError (undefined method `blank?' for nil:NilClass)
```